core/services/ocr2/plugins/ocr2keeper/evm21/block_subscriber.go

```diff
@@ -261,23 +210,49 @@ func (hw *BlockSubscriber) Unsubscribe(subId int) error {
 	return nil
 }
 
+func (hw *BlockSubscriber) processHead(h *evmtypes.Head) {
+	hw.mu.Lock()
+	defer hw.mu.Unlock()
+	// head parent is a linked list with EVM finality depth
+	// when re-org happens, new heads will have pointers to the new blocks
+	for cp := h; cp != nil; cp = cp.Parent {
```

Comment:
nit. are we guaranteed to reach a nil parent after finality depth? Maybe safer to also exit after initialBlockSize iterations?

Reply:
done